File: backend/app/schemas/mcp.py
"""MCP 连接器 Schema"""
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field
from datetime import datetime


class MCPConnectorCreate(BaseModel):
    name: str = Field(..., max_length=100)
    description: Optional[str] = ""
    transport: str = Field("sse", description="sse/stdio/http")
    url: Optional[str] = ""
    command: Optional[str] = ""
    args: Optional[List[str]] = None
    env_vars: Optional[Dict[str, str]] = None
    is_active: bool = True


class MCPConnectorUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    transport: Optional[str] = None
    url: Optional[str] = None
    command: Optional[str] = None
    args: Optional[List[str]] = None
    env_vars: Optional[Dict[str, str]] = None
    is_active: Optional[bool] = None


class MCPConnectorResponse(BaseModel):
    id: int
    name: str
    description: Optional[str] = ""
    transport: str
    url: Optional[str] = ""
    command: Optional[str] = ""
    args: Optional[List[str]] = None
    env_vars: Optional[Dict[str, str]] = None
    status: str
    tools_count: int = 0
    tools_list: Optional[List[Dict[str, Any]]] = None
    last_connected_at: Optional[datetime] = None
    error_message: Optional[str] = ""
    is_active: bool = True
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class MCPConnectorListResponse(BaseModel):
    total: int
    items: List[MCPConnectorResponse]


class MCPConnectResult(BaseModel):
    success: bool
    message: str
    tools_count: int = 0
    tools: List[Dict[str, Any]] = []

File: aits_hub_web/src/api/mcp.ts
import request from './request'

export interface MCPConnector {
  id: number
  name: string
  description: string
  transport: 'sse' | 'stdio' | 'http'
  url: string
  command: string
  args: string[]
  env_vars: Record<string, string>
  status: string
  tools_count: number
  tools_list: Array<{ name: string; description: string }>
  last_connected_at: string | null
  error_message: string
  is_active: boolean
  created_at: string
}

export interface MCPConnectorList {
  total: number
  items: MCPConnector[]
}

export const mcpApi = {
  list: (params?: { page?: number; page_size?: number; status?: string; keyword?: string }) =>
    request.get<MCPConnectorList>('/mcp/connectors', { params }),

  create: (data: Partial<MCPConnector>) =>
    request.post<MCPConnector>('/mcp/connectors', data),

  update: (id: number, data: Partial<MCPConnector>) =>
    request.put<MCPConnector>(`/mcp/connectors/${id}`, data),

  remove: (id: number) =>
    request.delete(`/mcp/connectors/${id}`),

  connect: (id: number) =>
    request.post<{ success: boolean; message: string; tools_count: number; tools: any[] }>(`/mcp/connectors/${id}/connect`),

  disconnect: (id: number) =>
    request.post(`/mcp/connectors/${id}/disconnect`),

  getTools: (id: number) =>
    request.get<{ tools: any[]; count: number }>(`/mcp/connectors/${id}/tools`),

  listAllTools: () =>
    request.get<{ total: number; tools: any[] }>('/mcp/tools'),
}
